micro-rdk-installer/src/main.rs

```diff
@@ -167,27 +172,52 @@ fn flash(binary_path: &str, should_monitor: bool) -> Result<(), Error> {
     Ok(())
 }
 
+fn get_path_or_download(path: Option<String>, tmp_dir: &TempDir) -> Result<PathBuf, Error> {
+    Ok(match path {
+        Some(path) => PathBuf::from(path),
+        None => {
+            let rt = Runtime::new().map_err(Error::AsyncError)?;
+            // let tmp_dir = Builder::new().prefix("micro-rdk-bin").tempdir().map_err(Error::FileError)?;
+            rt.block_on(download_latest_release(tmp_dir))?
+        }
+    })
+}
+
 fn main() -> Result<(), Error> {
     let cli = Cli::parse();
+    let tmp_dir = Builder::new()
+        .prefix("micro-rdk-bin")
+        .tempdir()
+        .map_err(Error::FileError)?;
     match &cli.command {
         Some(Commands::WriteCredentials(args)) => {
             let nvs_data = create_nvs_partition_binary(args.config.to_string(), args.nvs_size)?;
             write_credentials_to_app_binary(
-                &args.binary_path,
+                get_path_or_download(Some(args.binary_path.clone()), &tmp_dir)?,
                 &nvs_data,
                 args.nvs_size as u64,
                 args.nvs_offset,
             )?;
         }
         Some(Commands::WriteFlash(args)) => {
             let nvs_data = create_nvs_partition_binary(args.config.to_string(), args.nvs_size)?;
-            write_credentials_to_app_binary(
-                &args.binary_path,
+            let path = get_path_or_download(args.binary_path.clone(), &tmp_dir)?;
+            match write_credentials_to_app_binary(
+                path.clone(),
                 &nvs_data,
                 args.nvs_size as u64,
                 args.nvs_offset,
-            )?;
-            flash(&args.binary_path, args.monitor)?;
+            ) {
+                Ok(_) => {}
+                Err(e) => {
+                    if args.debug {
+                        std::fs::rename(path.clone(), "bad_binaries/micro-rdk-esp32-server.bin")
```

Comment:
what the purpose of bad_binaries?

Reply:
I wanted to give the user a way to examine the downloaded binary if there was something wrong with it and still clean up the file afterwards. But now that I think about it, they could just download from the assets page themselves